skbio/stats/composition.py

```diff
@@ -2058,3 +2060,489 @@ def dirmult_ttest(
+    # Implementation based on https://github.com/mortonjt/differential/blob/65752567ef4cf303471405b0a9be503eb10a0bbb/differential/util.py#L4
+    # TODO: Will need to improve this, as this is a very hacky solution.
+    for c in df.columns:
+        s = df[c]
+        try:
+            df[c] = s.astype(np.float64)
+        except Exception:
+            continue
+    return df
+
+
+def _lme_call(
+    formula,
+    metadata,
+    table,
+    groups,
+    reml=True,
+    method=None,
+    re_formula=None,
+    vc_formula=None,
+    fit_kwargs={},
+    **kwargs,
+):
+    # TODO: add docs when this implementation is approved
+    # _covariate_list is essentially the list of covariates
+
+    FEATUREID = "FeatureID"
+    LOG2FC = "Log2(FC)"
+    CI25 = "CI(2.5)"
+    CI975 = "CI(97.5)"
+    PVALUE = "pvalue"
+    COVARIATE = "Covariate"
+
+    submodels = []
+    metadata = _type_cast_to_float(metadata.copy())
+
+    merged_data = pd.merge(table, metadata, left_index=True, right_index=True)
+    if len(merged_data) == 0:
+        raise ValueError(
+            (
+                "No more samples left. Check to make sure that "
+                "the sample names between `metadata` and `data` "
+                "are consistent."
+            )
+        )
+
+    design_matrix = dmatrix(formula, metadata, return_type="dataframe")
+
+    # Obtaining the list of covariates by selecting the relevant columns
+    _covariate_list = design_matrix.columns.tolist()
+
+    # Removing intercept since it is not a covariate, and is included by default
+    _covariate_list.remove("Intercept")
+
+    output = []
+    for response_var in table.columns:
+        # mixed effects code is obtained here:
+        # http://stackoverflow.com/a/22439820/1167475
+        stats_formula = "%s ~ %s" % (response_var, formula)
+        model = MixedLM.from_formula(
+            formula=stats_formula,
+            data=merged_data,
+            groups=groups,
+            re_formula=re_formula,
+            vc_formula=vc_formula,
+            **kwargs,
+        )
+
+        results = model.fit(reml=reml, method=method, **fit_kwargs)
+        summary = results.summary()
+
+        for var_name in _covariate_list:
+            try:
+                _lme_coef, _lme_25, _lme_975 = np.nan, np.nan, np.nan
+                _lme_coef = float(summary.tables[1]["Coef."][var_name])
+                _lme_25 = float(summary.tables[1]["[0.025"][var_name])
+                _lme_975 = float(summary.tables[1]["0.975]"][var_name])
+
+                individual_results = {
+                    FEATUREID: response_var,
+                    COVARIATE: var_name,
+                    LOG2FC: _lme_coef,
+                    CI25: _lme_25,
+                    CI975: _lme_975,
+                    PVALUE: results.pvalues[var_name],
+                }
+            except Exception as e:
+                if type(e) is not ValueError:
+                    print("Something went really wrong, details below: ")
```

Comment:
This could give users a negative reaction. Let's rename this to 

```suggestion
                    print(f"LME fit failed for covariate {var_name} and `response_var`, outputting nans.")
```

Reply:
Thank you so much for this suggestion! It does describes the error in a much better way.